tests/testthat/test-unnest.R

```diff
@@ -0,0 +1,289 @@
+  expect_equal(out, tibble(x = integer(), y = unspecified()))
+})
+
+test_that("unnesting bare lists of NULLs is equivalent to unnesting empty lists", {
+  df <- tibble(x = 1L, y = list(NULL))
+  out <- unnest(df, y)
+  expect_identical(out, tibble(x = integer(), y = unspecified()))
+})
+
+test_that("unnest() preserves ptype", {
+  tbl <- tibble(x = integer(), y = list_of(ptype = tibble(a = integer())))
+  res <- unnest(tbl, y)
+  expect_equal(res, tibble(x = integer(), a = integer()))
+})
+
+test_that("unnesting typed lists of NULLs retains ptype", {
+  df <- tibble(x = 1L, y = list_of(NULL, .ptype = tibble(a = integer())))
+  out <- unnest(df, y)
+  expect_identical(out, tibble(x = integer(), a = integer()))
+})
+
+test_that("ptype can be overriden manually (#1158)", {
+  df <- tibble(
+    a = list("a", c("b", "c")),
+    b = list(1, c(2, 3)),
+  )
+
+  ptype <- list(b = integer())
+
+  out <- unnest(df, c(a, b), ptype = ptype)
+
+  expect_type(out$b, "integer")
+  expect_identical(out$b, c(1L, 2L, 3L))
+})
+
+test_that("ptype works with nested data frames", {
+  df <- tibble(
+    a = list("a", "b"),
+    b = list(tibble(x = 1, y = 2L), tibble(x = 2, y = 3L)),
+  )
+
+  # x: double -> integer
+  ptype <- list(b = tibble(x = integer(), y = integer()))
+
+  out <- unnest(df, c(a, b), ptype = ptype)
+
+  expect_identical(out$x, c(1L, 2L))
+  expect_identical(out$y, c(2L, 3L))
+})
+
+test_that("skips over vector columns", {
+  df <- tibble(x = integer(), y = list())
+  expect_identical(unnest(df, x), df)
+})
+
+test_that("unnest keeps list cols", {
+  df <- tibble(x = 1:2, y = list(3, 4), z = list(5, 6:7))
+  out <- df %>% unnest(y)
+
+  expect_equal(names(out), c("x", "y", "z"))
+})
+
+# Deprecated behaviours ---------------------------------------------------
+
+test_that("warn about old style interface", {
+  df <- tibble(x = c(1, 1, 1), y = 1:3)
+
+  expect_snapshot(out <- nest(df, y))
+  expect_named(out, c("x", "data"))
+
+  expect_snapshot(out <- nest(df, -y))
+  expect_named(out, c("y", "data"))
+})
+
+test_that("only warn about unnamed inputs (#1175)", {
+  df <- tibble(x = 1:3, y = 1:3, z = 1:3)
+  expect_snapshot(out <- nest(df, x, y, foo = z))
+  expect_named(out, c("foo", "data"))
+})
+
+test_that("unnamed expressions are kept in the warning", {
+  df <- tibble(x = 1:3, z = 1:3)
+  expect_snapshot(out <- nest(df, x, starts_with("z")))
+  expect_named(out, "data")
+})
+
+test_that("can control output column name", {
+  df <- tibble(x = c(1, 1, 1), y = 1:3)
+  expect_snapshot(out <- nest(df, y, .key = "y"))
+  expect_named(out, c("x", "y"))
+})
+
+test_that("can control output column name when nested", {
+  df <- dplyr::group_by(tibble(x = c(1, 1, 1), y = 1:3), x)
+  expect_snapshot(out <- nest(df, .key = "y"))
+  expect_named(out, c("x", "y"))
+})
+
+test_that(".key gets warning with new interface", {
+  df <- tibble(x = c(1, 1, 1), y = 1:3)
+  expect_snapshot(out <- nest(df, y = y, .key = "y"))
+  expect_named(df, c("x", "y"))
+})
```

Comment:
This chunk should stay with `nest()` tests + regenerate snapshots in this file and in `test-nest.R`